tuplex/core/src/HistoryServerConnector.cc

```diff
@@ -222,6 +255,25 @@ namespace tuplex {
 //                _reservoirs.emplace_back(reservoir);
 //            }
 //        });
+        assert(plan);
+
+        // go through each stage
+        plan->foreachStage([this](const PhysicalStage* stage) {
+            assert(stage);
+
+            // is trafo stage?
+            const TransformStage* tstage = nullptr;
+            if(tstage = dynamic_cast<const TransformStage*>(stage)) {
+                auto operators = tstage->get_ops();
+                if(operators.empty())
+                    return;
+                auto reservoir = std::make_shared<TransformStageExceptionReservoir>(tstage, operators, _exceptionDisplayLimit);
+
+                for(auto& op : operators)
```

Comment:
const auto& (getID should be const function)

Reply:
Done.